### PR TITLE
Parse seeds same as java

### PR DESCRIPTION
**Pull request by kralverde**
## Description
This PR implements parsing a seed from a string the same as Java.

## Testing
N/A

## Checklist
Things need to be done before this Pull Request can be merged.

- [x] Code is well-formatted and adheres to project style guidelines: `cargo fmt`
- [x] Code does not produce any clippy warnings: `cargo clippy`
- [x] All unit tests pass: `cargo test`

### CODE REVIEW

**Comment by Snowiiii** [comment]
Hey can you may tell from what Vanilla java classes you have these magic numbers from ?

**Comment by kralverde** [comment]
> Hey can you may tell from what Vanilla java classes you have these magic numbers from ?

Random.getSeed()

**Comment by kralverde** [comment]
Theoretically, it can just be any random u64 as a seed; we don't really need to use MC's implementation.

File: pumpkin-core/src/random/mod.rs [hunk 1]
```diff
@@ -1,10 +1,35 @@
...
+        .unwrap();
+
+    let nanos = time::SystemTime::now()
+        .duration_since(time::SystemTime::UNIX_EPOCH)
+        .unwrap()
+        .as_nanos();
+
+    let nano_upper = (nanos >> 8) as u64;
+    let nano_lower = nanos as u64;
+    seed ^ nano_upper ^ nano_lower
```
Comment (Snowiiii) [inline pumpkin-core/src/random/mod.rs:30]: You can/should use something similar to java's compareAndSet. You can look at https://github.com/kallekankaanpaa/jandom/blob/master/src/lib.rs#L228

Comment (kralverde) [reply pumpkin-core/src/random/mod.rs:30]: That's what fetch_update does I thought

**Comment by Snowiiii** [comment]
Overall looks good. 
Thanks @kralverde

**Pull request merged**